components/script/dom/transformstream.rs

```diff
@@ -436,6 +440,68 @@ impl TransformStream {
         )
     }
 
+    /// Creates and set up the newly created transform stream following
+    /// <https://streams.spec.whatwg.org/#transformstream-set-up>
+    pub(crate) fn set_up(
+        &self,
+        cx: SafeJSContext,
+        global: &GlobalScope,
+        cancel: Option<TransformerCancelAlgorithmType>,
+        flush: Option<TransformerFlushAlgorithmType>,
+        transform: TransformerTransformAlgorithmType,
+        can_gc: CanGc,
+    ) -> Fallible<()> {
+        // Step1. Let writableHighWaterMark be 1.
+        let writable_high_water_mark = 1.0;
+
+        // Step 2. Let writableSizeAlgorithm be an algorithm that returns 1.
+        let writable_size_algorithm = extract_size_algorithm(&Default::default(), can_gc);
+
+        // Step 3. Let readableHighWaterMark be 0.
+        let readable_high_water_mark = 0.0;
+
+        // Step 4. Let readableSizeAlgorithm be an algorithm that returns 1.
+        let readable_size_algorithm = extract_size_algorithm(&Default::default(), can_gc);
+
+        // Step 5. Let transformAlgorithmWrapper be an algorithm that runs these steps given a value chunk:
+        // Step 6. Let flushAlgorithmWrapper be an algorithm that runs these steps:
+        // Step 7. Let cancelAlgorithmWrapper be an algorithm that runs these steps given a value reason:
+        // NOTE: These steps are implemented in `TransformStreamDefaultController`
+
+        // Step 8. Let startPromise be a promise resolved with undefined.
+        let start_promise = Promise::new_resolved(global, cx, (), can_gc);
+
+        // Step 9. Perform ! InitializeTransformStream(stream, startPromise,
+        // writableHighWaterMark, writableSizeAlgorithm, readableHighWaterMark,
+        // readableSizeAlgorithm).
+        self.initialize(
+            cx,
+            global,
+            start_promise.clone(),
+            writable_high_water_mark,
+            writable_size_algorithm,
+            readable_high_water_mark,
+            readable_size_algorithm,
+            can_gc,
+        )?;
+
+        // Step 10. Let controller be a new TransformStreamDefaultController.
+        let controller = TransformStreamDefaultController::new_with_algorithms(
```

Comment:
This should be rather `TransformStreamDefaultController::new()`, where `transformer: &Transformer` is replaced with a `TransformerType` enum.